### ISSUE TITLE
What is the relationship between the output label of pretrained model in model zoo and wordnet synset id? 

### ISSUE THREAD

**Issue by imkzh** [pytorch/vision#484]
we can easily access pytorch pre-trained model like VGG, AlexNet and SqueezeNet by

    import torchvision 
    torchvision.models.vgg16(pretrained=True)

can anyone point out what's the relationship between the output label(index of maximum output value) and the actual category?

i downloaded ILSVRC2012_devkit_t12 and got the imagenet id and other metainfo provided by meta.mat, however it seems pre-trained model have some different id. because when i evaluate the network with ILSVRC2012 validation set, it reports 100% error.

**Comment by Cadene** [comment]
You can find an example [here](https://github.com/Cadene/pretrained-models.pytorch/blob/master/examples/imagenet_logits.py).

You will need two files:
- [imagenet_classes.txt](https://github.com/Cadene/pretrained-models.pytorch/blob/master/data/imagenet_classes.txt)
- [imagenet_synsets.txt](https://github.com/Cadene/pretrained-models.pytorch/blob/master/data/imagenet_synsets.txt)

**Comment by imkzh** [comment]
Thank you for your help, they are exactly what i need.

**Comment by fmassa** [comment]
I think we should provide somewhere the mapping between the class names and the indices. Not sure if it should be in `pytorch/examples` or in torchvision. What do you think?

**Comment by imkzh** [comment]
That's great! i think it will be torchvision to provide those mapping, and `pytorch/examples` to have some example on it. :grinning: am i too greedy?

**Comment by fmassa** [comment]
I think this totally make sense to provide this mapping.
I'll think about a good way of doing it, maybe making a base class `TorchvisionModel`, that contains the basic pre-processing parameters that are used for all the models, as well as the image synset.

**Comment by theoptips** [comment]
+1 would appreciate a model.class_ method or model.labels_ kind of like sklearn. Developers can easily see the mapping between output labels and index. May be this feature already exists? Would love to know how to do it. Thanks!

**Comment by NizarTarabay** [comment]
> Thank you for your help, they are exactly what i need, as torchvision modules itself didn't provide any documentations on this.
How did you manage to make it work, even with these two text files there still a discrepancy between the output predicted labels and the imagenet validation dataset.
I am using Squeezenet to predict the validation classes on ILSVRC2012. The pretrained model seems working well, however, the accuracy is 0!

**Comment by stared** [comment]
I think that class names (i.e.  a map `idx -> class name`) are an integral part of any pre-trained classifier.  Would a PR with class names be welcome?
